punishments.js

```diff
@@ -408,6 +506,81 @@ Punishments.banRange = function (range, reason) {
 	Punishments.ips.set(range, punishment);
 };
 
+Punishments.roomBan = function (room, user, noRecurse, userid) {
+	if (!userid) userid = user.userid;
+	let alts;
+	let keys = noRecurse;
+	if (!keys || !keys.add) {
+		keys = new Set();
+	}
+	if (!noRecurse) {
+		alts = [];
+		keys = new Set();
+		Users.users.forEach(otherUser => {
+			if (otherUser === user) return;
+			for (let myIp in user.ips) {
+				if (myIp in otherUser.ips) {
+					alts.push(otherUser.name);
+					Punishments.roomBan(room, otherUser, keys, userid);
+					return;
+				}
+			}
+		});
+	}
+
+	let punishment = ['ROOMBAN', userid, Date.now() + ROOMBAN_DURATION];
+
+	Punishments.roombannedUserids.set(room.id + '|' + userid, punishment);
+	if (user.autoconfirmed) {
+		keys.add(room.id + '|' + user.autoconfirmed);
+		Punishments.roombannedUserids.set(room.id + '|' + user.autoconfirmed, punishment);
+	}
+	if (room.game && room.game.removeBannedUser) {
+		room.game.removeBannedUser(user);
+	}
+	for (let ip in user.ips) {
+		Punishments.roombannedIps.set(room.id + '|' + ip, punishment);
+		keys.add(room.id + '|' + ip);
+	}
+	if (!user.can('bypassall')) user.leaveRoom(room.id);
+	if (!noRecurse) {
+		keys.delete(punishment[1]);
+		Punishments.appendPunishment({
+			keys: Array.from(keys),
+			punishType: punishment[0],
+			rest: punishment.slice(2),
+		}, punishment[1], ROOMBANS_FILE);
+	}
+	return alts;
+};
+
+Punishments.unRoomBan = function (room, userid, noRecurse) {
```

Comment:
If you pay attention to the conventions in the rest of this file, it'd be `roomban` and `unroomban`...